### DISCUSSION TITLE
Give the ability to skip or ignore SSO on an organization's public repo

### DISCUSSION THREAD

**Discussion by Aymendje** [community/community#9045] [Projects and Issues]
My personal github account is linked to some of my organization repos.
To sign in with my organization privileges, I need to be connected through VPN (or their SSO will reject me)
But if I am connected on my personal github and I want to see a public repo owned by my organization, it directly prompts me the SSO login page and I cannot skip it. The only way to access the page is through incognito / disconnect from my Github account.

Here is an example to be clear :

aymendje (my personal github account) is linked to the Microsoft organization (that requires 2FA / SSO)
When I try to access a page like https://github.com/microsoft/vscode/releases/, it forces me to sign in 
<img width="285" alt="image" src="https://user-images.githubusercontent.com/5965677/146632768-77a0a54a-3a9a-4e88-adf8-793c27a399e6.png">

But since I am on my personal device / not connected to VPN / Corpnet, my 2FA gets rejected and I am stuck in the Microsoft's SSO Error page (you are not authorized)

Ideally, we would have a button "view this page outside my organization" or simply "skip" alongside the "continue" button, and view the page without signing in to the organization.

I guess that I should have 2 github accounts, a professional one, and a work one, but its so much simpler to use only one and leverage the organizations instead !

**Comment by Ashesh3** [comment]
How is this still a thing in 2024? I have the exact same issue. I do not want to escalate my privileges by using single sign-on for a public repo and instead want to interact with it with my personal account. It's really troublesome to open an incognito window just to access a public repo.

**Comment by thesofakillers** [comment]
bump

**Comment by mark-wiemer** [comment]
Wild that this is still happening. I'm just opening these pages in InPrivate tabs for now, a terrible workaround :(

**Reply by mark-wiemer** [reply]
I've since disconnected my personal GitHub from my work one, always fun to have two accounts for the same purpose :(

**Comment by sethupavan12** [comment]
bump. 
It's been almost 3 years with no update or ETA for a fix.

**Comment by agg23** [comment]
Expressing my frustration with this as well. It seems like a number of Microsoft employees in particular are being hit with this, given the responses in this thread.

**Comment by SahilAshar** [comment]
gentle bump - would love a workaround for this :)

**Reply by SahilAshar** [reply]
Update: MSFT has since gone through and done a manual disconnect for my account (not sure when it occurred, or if it's occurred for others). I am no longer running into this issue, though I do feel that GitHub should still offer users a way to manually disconnect this themselves.

**Reply by agg23** [reply]
This is not true.

**Reply by SahilAshar** [reply]
@agg23 - apologies, to clarify, MSFT has manually disconnected my account. I'm not sure of the state of others. I'll update my reply to avoid further confusion :)

**Comment by priyanshu92** [comment]
Bump. I am getting this on my personal system when trying to download Powershell from GitHub releases. This is ridiculous.

![image](https://github.com/user-attachments/assets/cee50c32-2bc1-4b81-a60c-70fbe1e77164)

**Comment by trupples** [comment]
bump

**Comment by Ashesh3** [comment]
How is this still a thing in 2025?

**Comment by pinkfloydx33** [comment]
Can't report issues, can't download releases... and my personal (only) github account is not linked to an organization either. I am very baffled by this... what's going on?

**Comment by puneetred-cdx** [comment]
bump

**Comment by froddd** [comment]
This is still very much an issue, and given the [GitHub documentation on SAML/SSO](https://docs.github.com/en/enterprise-cloud@latest/authentication/authenticating-with-single-sign-on/about-authentication-with-single-sign-on#about-authentication-with-sso), I believe this should be considered a bug:

> IdP authentication is not required for accessing public repositories in certain ways:
> 
> - Viewing the repository's overview page and file contents on GitHub
> - Forking the repository
> - Performing read operations via Git, such as cloning the repository
> 
> Authentication is required for other access to public repositories, such as viewing issues, pull requests, projects, and releases.

Clearly, IdP authentication cannot be bypassed, and as such _is_ required in practice for accessing public repositories in the ways described above. It should not be.

The suggestion of a link to “skip” or “View as” would work well.

**Comment by datah4zard** [comment]
ba-bump.
